src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -384,6 +387,9 @@ private void CountDownChildTest()
             _childTestCountdown.Signal();
             if (_childTestCountdown.CurrentCount == 0)
             {
+                if (Test.TestType == "Theory" && Result.ResultState == ResultState.Success && Result.PassCount == 0)
```

Comment:
I wish we could stop doing the `TestType == "Theory"` check (tell-don't-ask) sometime in the future. This is the third one being added.